src/Integration/IntegrationRegistry.php

```diff
@@ -126,6 +126,20 @@ private function getDefaultIntegrations(Options $options): array
             return [];
         }
 
+        /**
+         * If the DSN is null, we do not want to register any eror handler.
+         * Hence, we do not setup the ExceptionListenerIntegration, ErrorListenerIntegration and
+         * FatalErrorListenerIntegration in this case.
+         */
+        if (null === $options->getDsn()) {
+            return [
+                new RequestIntegration(),
+                new TransactionIntegration(),
+                new FrameContextifierIntegration(),
+                new EnvironmentIntegration(),
+            ];
```

Comment:
Mainly did this for better readability and less complexity.

Reply:
I'm not convinced that having to maintain two separated lists instead of just splitting its creation is really changing so much the readability, but it's your call